src/coreclr/jit/redundantbranchopts.cpp

```diff
@@ -62,6 +62,17 @@ PhaseStatus Compiler::optRedundantBranches()
         block->bbFlags &= ~BBF_VISITED;
     }
 
+    // We've made some changes so it makes sense to try
+    // one more time
+    if (visitor.madeChanges)
+    {
+        visitor.WalkTree();
```

Comment:
I wonder if we can scope this down better somehow. Rewalking every block likely leads to a lot of redundant failures. We are paying nearly 1% in TP for this version.

One simple thing (which probably doesn't do much for TP) is that up in PostOrderVisit we can skip any block that has zero refs.

But I wonder if there's some other way to deduce which blocks likely would benefit (or conversely, likely would not benefit) from reconsideration.

There is fairly extensive literature on incremental dominator tree maintenance, for example
* [Dynamic Dominators and Low-High Orders in DAGs](https://drops.dagstuhl.de/opus/volltexte/2019/11171/pdf/LIPIcs-ESA-2019-50.pdf)
* [Incremental Computation of Dominator Trees](https://dl.acm.org/doi/pdf/10.1145/202529.202531)

so perhaps there's some clue in there about where revisiting might be fruitful.

Reply:
Also, the dominator tree itself is not accurate. 

I think this just leads to missed opportunities as during the rewalk as we might visit blocks in the wrong order (we are trying for bottom-up). But I'm not 100% sure.